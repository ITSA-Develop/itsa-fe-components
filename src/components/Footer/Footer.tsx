export interface IFooterProps {
	version: string;
	text: string;
}

export const Footer = ({ version, text }: IFooterProps) => {
	return (
		<footer className="fixed bottom-0 left-0 w-full text-black-100 py-4 flex items-center justify-center text-sm gap-3">
			<p className="text-center">{text}</p>
			<p className="text-center">{version}</p>
		</footer>
	);
};
