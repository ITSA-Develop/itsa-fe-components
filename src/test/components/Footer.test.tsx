import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Footer, IFooterProps } from '../../components/Footer/Footer';

describe('Footer component', () => {
	const defaultProps: IFooterProps = {
		version: 'v1.2.3',
		text: 'Copyright 2025',
	};

	it('renders the text and version correctly', () => {
		const { container } = render(<Footer {...defaultProps} />);

		expect(screen.getByText(defaultProps.text)).toBeInTheDocument();

		expect(screen.getByText(defaultProps.version)).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
