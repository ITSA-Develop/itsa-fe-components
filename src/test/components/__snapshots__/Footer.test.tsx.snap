// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer component > renders the text and version correctly 1`] = `
<div>
  <footer
    class="fixed bottom-0 left-0 w-full text-black-100 py-4 flex items-center justify-center text-sm gap-3"
  >
    <p
      class="text-center"
    >
      Copyright 2025
    </p>
    <p
      class="text-center"
    >
      v1.2.3
    </p>
  </footer>
</div>
`;
